### PR TITLE
Improve stream analyze configuration

### PR DESCRIPTION
**Pull request by oleh-poberezhets**
Closes [97](https://github.com/LCMApps/video-quality-tools/issues/97)

### CODE REVIEW

File: src/FramesMonitor.js [hunk 1]
```diff
@@ -70,14 +71,19 @@ class FramesMonitor extends EventEmitter {
             throw new Errors.ConfigError('exitProcessGuardTimeoutInMs param should be a positive integer.');
         }
 
+        if (analyzeDurationInMs !== undefined && (!_.isSafeInteger(analyzeDurationInMs) || analyzeDurationInMs <= 0)) {
+            throw new Errors.ConfigError('You should provide a correct analyze duration, bastard.');
```
Comment (WoZ) [inline src/FramesMonitor.js]: Don't use rude lexical terms, please =)

Btw, I propose to not force a user to set this option. We may set it by default to 5 sec? As I remember, it's a default value for the ffmpeg.

Comment (WoZ) [reply src/FramesMonitor.js]: Oops, 10 lines below I see the code that sets this default value.

File: src/StreamsInfo.js [hunk 2]
```diff
@@ -31,11 +31,16 @@ class StreamsInfo {
             throw new Errors.ConfigError('You should provide a correct timeout, bastard.');
         }
 
+        if (analyzeDurationInMs !== undefined && (!_.isInteger(analyzeDurationInMs) || analyzeDurationInMs <= 0)) {
+            throw new Errors.ConfigError('You should provide a correct analyze duration, bastard.');
```
Comment (WoZ) [inline src/StreamsInfo.js]: The same rude lexicon.

File: tests/Unit/FramesMonitor/constructor.test.js [hunk 3]
```diff
@@ -177,7 +177,8 @@ describe('FramesMonitor::constructor', () => {
             bufferMaxLengthInBytes     : config.bufferMaxLengthInBytes,
             errorLevel                 : config.errorLevel,
             exitProcessGuardTimeoutInMs: config.exitProcessGuardTimeoutInMs,
-            timeout                    : config.timeoutInMs * 1000
+            timeout                    : config.timeoutInMs * 1000,
+            analyzeDuration            : config.analyzeDurationInMs && config.analyzeDurationInMs * 1000 || 0
```
Comment (WoZ) [inline tests/Unit/FramesMonitor/constructor.test.js]: No logic in test data. Test data must be clear and without any variance and side effects. So, it's better to specify it explicitly in the `config`.

File: src/StreamsInfo.js (hunk 2 above)
Comment (oleh-poberezhets) [reply src/StreamsInfo.js]: fixed

File: tests/Unit/FramesMonitor/constructor.test.js (hunk 3 above)
Comment (oleh-poberezhets) [reply tests/Unit/FramesMonitor/constructor.test.js]: was improved

**Comment by oleh-poberezhets** [comment]
> Btw, is it possible to write a functional test to confirm the correctness of its logic? Like the test in `listen.test.js`

It is hard to reproduce with a static file. This option is useful for live streaming.

**Pull request merged**